### PR TITLE
chore(lockfile): update on snyk auto PRs

### PR DESCRIPTION
**Pull request by ken-mellem**
## Notes

I do not that author = 'snyk-bot' when looking at git history.

### CODE REVIEW

File: .github/workflows/update-lockfile.yml [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
         uses: actions/checkout@v3
 
       - name: Update lockfile
-        run: yarn install
+        uses: ./.github/actions/client-build       
+
+      - name: Check if yarn.lock has changed
+        id: check-yarn-lock
+        run: |
+          git add yarn.lock
```
Comment (ken-mellem) [inline .github/workflows/update-lockfile.yml:19]: should this be `./client/yarn.lock` ?

Comment (Noggling) [reply .github/workflows/update-lockfile.yml:19]: yeah i think it would be better.

File: .github/workflows/client-pr.yml [hunk 2]
```diff
@@ -9,6 +9,12 @@ on:
     types: [opened, synchronize]
 
 jobs:
+  # SNYK auto PRs does not always update yarn.lock file
+  check-lock-file:
+    name: Update yarn.lock for Snyk updates
+    if: startsWith(github.head_ref, 'snyk-')
+    uses: ./.github/workflows/update-lockfile.yml
```
Comment (Noggling) [inline .github/workflows/client-pr.yml:16]: you cant call a workflow. we need to write it as a custom action.

move file to a folder called update lock file.
add a file called action.yml

File: .github/workflows/update-lockfile.yml [hunk 3]
```diff
@@ -1,27 +1,34 @@
 name: Update lockfile
 
 on:
-  workflow_dispatch:
-  pull_request:
-    branches:
-      - main
+  workflow_call:  # Called in the client-pr workflow
```
Comment (Noggling) [inline .github/workflows/update-lockfile.yml:4]: this file should not be a workflow

**Comment by Noggling** [comment]
Open again when ready!

**Pull request closed without merging**